nvflare/app_common/job_schedulers/job_scheduler.py

```diff
@@ -19,10 +19,20 @@
 from nvflare.apis.fl_component import FLComponent
 from nvflare.apis.fl_constant import FLContextKey
 from nvflare.apis.fl_context import FLContext
-from nvflare.apis.job_def import Job, JobMetaKey
+from nvflare.apis.job_def import ALL_SITES, Job, JobMetaKey
 from nvflare.apis.job_scheduler_spec import DispatchInfo, JobSchedulerSpec
 from nvflare.apis.server_engine_spec import ServerEngineSpec
 
+SERVER_SITE_NAME = "server"
+
+
+def _check_special_deploy_map(deploy_map):
+    # deploy_map: {"app_name": ["ALL_SITES"]} will be treated as deploying to all online clients
+    all_keys = list(deploy_map.keys())
+    if len(all_keys) == 1 and len(deploy_map[all_keys[0]]) == 1 and deploy_map[all_keys[0]][0].upper() == ALL_SITES:
```

Comment:
all_keys length is not always 1. For example,

{
  "deploy_map": {
    "app1": ["@ALL"],
    "app2": []
  }
}